### DISCUSSION TITLE
Conditional pipe execution?

### DISCUSSION THREAD

**Discussion by JoshC8C7** [explosion/spaCy#7680] [Help: Coding & Implementations] [answered]
Is there support for conditional execution of a pipeline component, e.g. only run NER if a previous custom component has set `doc._.runNER` or similar. At the minute I'm just running `nlp.pipe` and then performing the check as to whether NER is required via just running a pipe-like function, and then running `EntityRecognizer` separately if so - this then provides a bottleneck as I'm just running individual components on a loop rather than benefiting from `pipe` though.

This may not be functionality or may just not be able to find it.
Thanks

**Comment by polm** [comment, marked as answer]
No, this functionality doesn't exist. 

You might be able to implement it by subclassing a built-in pipeline and adding the check that way, though I am not sure it would interact correctly with pipe, which I think assumes you can just batch up all the incoming documents.
